src/log.rs

```diff
@@ -7,7 +7,7 @@ macro_rules! info {
     };
     ($ctx:expr, $msg:expr, $($args:expr),* $(,)?) => {{
         let formatted = format!($msg, $($args),*);
-        let full = format!("{file}:{line}: {msg}",
+        let full = format!("{file:21}:{line:4}: {msg}",
```

Comment:
I don't understand, where do these numbers come from? Maybe put a comment

Reply:
I included this here because it's not worth to make an extra pr for that?!
Will provide a compare of both log formats. It's not really a change. It's only for better reading the logfile.

Reply:
short compare what the numbers in format string are doing (column width).

Without numbers:
```
08-26 07:11:22.841   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:702: 1 mails read from "INBOX".
08-26 07:11:22.841   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:594: Imap::fetch_new_messages
08-26 07:11:23.011   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:618: setting network_online => true
08-26 07:11:23.011   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:637: fetch_new_messages: ignoring uid 41311, last seen was 41311
08-26 07:11:23.011   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:702: 0 mails read from "INBOX".
08-26 07:11:23.011   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:118: INBOX-fetch done in 565 ms
08-26 07:11:23.021   581   622 I DeltaChat: ThreadId(5) src/job.rs:631: INBOX-IDLE will not be started because of waiting jobs.
08-26 07:11:23.021   581   622 I DeltaChat: ThreadId(5) src/job.rs:1043: perform_inbox_jobs: called
08-26 07:11:23.021   581   622 I DeltaChat: ThreadId(5) src/job.rs:1069: Imap job_perform: called  - probe_network false
08-26 07:11:23.021   581   622 I DeltaChat: ThreadId(5) src/job.rs:1071: Imap job #3, action MoveMsg loaded - tries 0 - action MoveMsg
08-26 07:11:23.021   581   622 I DeltaChat: ThreadId(5) src/job.rs:1077: Imap job #3, action MoveMsg - tries 0 - action MoveMsg started...
08-26 07:11:23.322   581   622 I DeltaChat: ThreadId(5) src/job.rs:1228: Imap perform_job_action: begin immediate try 0 of job #3, action MoveMsg
08-26 07:11:23.502   581   622 I DeltaChat: ThreadId(5) src/job.rs:1261: Imap finished immediate try 0 of job #3, action MoveMsg - try_res Finished
08-26 07:11:23.502   581   622 I DeltaChat: ThreadId(5) src/job.rs:1214: Imap remove job ##3, action MoveMsg as it succeeded
08-26 07:11:23.502   581   622 I DeltaChat: ThreadId(5) src/job.rs:1216: Imap job ##3, action MoveMsg, setting last_job_success => true
08-26 07:11:23.542   581   622 I DeltaChat: ThreadId(5) src/job.rs:1071: Imap job #4, action MarkseenMsgOnImap loaded - tries 0 - action MarkseenMsgOnImap
08-26 07:11:23.542   581   622 I DeltaChat: ThreadId(5) src/job.rs:1077: Imap job #4, action MarkseenMsgOnImap - tries 0 - action MarkseenMsgOnImap started...
08-26 07:11:23.842   581   622 I DeltaChat: ThreadId(5) src/job.rs:1228: Imap perform_job_action: begin immediate try 0 of job #4, action MarkseenMsgOnImap
08-26 07:11:23.842   581   622 I DeltaChat: ThreadId(5) src/job.rs:1261: Imap finished immediate try 0 of job #4, action MarkseenMsgOnImap - try_res Finished
08-26 07:11:23.842   581   622 I DeltaChat: ThreadId(5) src/job.rs:1214: Imap remove job ##4, action MarkseenMsgOnImap as it succeeded
08-26 07:11:23.842   581   622 I DeltaChat: ThreadId(5) src/job.rs:1216: Imap job ##4, action MarkseenMsgOnImap, setting last_job_success => true
08-26 07:11:23.922   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:76: JobThread::fetch INBOX-thread
08-26 07:11:23.922   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:103: JobThread::connect_and_fetch INBOX-thread
08-26 07:11:23.922   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:107: setting +++ network_online +++ => true
08-26 07:11:23.922   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:111: INBOX-fetch started...
08-26 07:11:23.922   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:477: Imap::fetch
08-26 07:11:23.922   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:594: Imap::fetch_new_messages
08-26 07:11:24.082   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:618: setting network_online => true
08-26 07:11:24.082   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:637: fetch_new_messages: ignoring uid 41310, last seen was 41311
08-26 07:11:24.082   581   622 I DeltaChat: ThreadId(5) src/imap/mod.rs:702: 0 mails read from "INBOX".
08-26 07:11:24.082   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:118: INBOX-fetch done in 163 ms
08-26 07:11:24.092   581   622 I DeltaChat: ThreadId(5) src/job_thread.rs:186: INBOX-IDLE started...
08-26 07:11:24.212   581   622 I DeltaChat: ThreadId(5) src/imap/idle.rs:91: Idle wait (Secure) - timeout set to 1380s s
```

With numbers:
```
09-21 14:44:20.800 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 193: INBOX-IDLE ended...
09-21 14:44:20.800 17847 17879 W DeltaChat: ThreadId(5)  src/job_thread.rs    : 195: INBOX-IDLE failed: IMAP IDLE protocol timed out -> reconnecting
09-21 14:44:20.800 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 349: IMAP unsetup_handle (disconnected)
09-21 14:44:20.930 17847 17879 I DeltaChat: ThreadId(5)  src/job.rs           :1065: perform_inbox_jobs: go ...
09-21 14:44:20.930 17847 17879 I DeltaChat: ThreadId(5)  src/job.rs           :1100: Imap job_perform: no job found => no action
09-21 14:44:20.930 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    :  76: JobThread::fetch INBOX-thread
09-21 14:44:20.930 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 104: JobThread::connect_and_fetch INBOX-thread
09-21 14:44:20.940 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 349: IMAP unsetup_handle (disconnected)
09-21 14:44:23.382 17847 17879 I DeltaChat: ThreadId(5)  src/context.rs       : 171: === network_online === is => true
09-21 14:44:23.392 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 113: INBOX-fetch started...
09-21 14:44:23.392 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 594: Imap::fetch_new_messages
09-21 14:44:23.892 17847 17879 I DeltaChat: ThreadId(5)  src/context.rs       : 171: === network_online === is => true
09-21 14:44:24.333 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs:  42: Receiving message INBOX/43292...
09-21 14:44:24.353 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs:1094: not creating ad-hoc group for mailing list message
09-21 14:44:24.353 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs: 419: Message belongs to a mailing list and is ignored.
09-21 14:44:24.613 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs: 649: Message has 1 parts and is assigned to chat #Chat#Trash.
09-21 14:44:24.613 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs: 215: received message 43292 has Message-Id: deltachat/deltachat-core-rust/pull/1939/review/492534132@github.com
09-21 14:44:24.673 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 703: 1 mails read from "INBOX".
09-21 14:44:24.673 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 594: Imap::fetch_new_messages
09-21 14:44:24.943 17847 17879 I DeltaChat: ThreadId(5)  src/context.rs       : 171: === network_online === is => true
09-21 14:44:24.943 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 638: fetch_new_messages: ignoring uid 43292, last seen was 43292
09-21 14:44:24.943 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 703: 0 mails read from "INBOX".
09-21 14:44:24.943 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 120: INBOX-fetch done in 1551 ms
09-21 14:44:24.943 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 191: INBOX-IDLE started...
09-21 14:44:25.133 17847 17879 I DeltaChat: ThreadId(5)  src/imap/idle.rs     :  92: Idle wait (Secure) - timeout set to 780s
09-21 14:44:25.133 17847 17879 I DeltaChat: ThreadId(5)  src/imap/idle.rs     : 103: Idle wait - entering wait-on-remote state
09-21 14:56:00.490 17847 19277 I DeltaChat: -------------------- FetchWorker.doWork() started --------------------
09-21 14:56:00.490 17847 19277 I DeltaChat: ThreadId(25) src/job.rs           : 672: interrupt_inbox_idle: called -> from FetchWorker.doWork()
09-21 14:56:00.490 17847 19277 I DeltaChat: ThreadId(25) src/job.rs           : 691: interrupt_inbox_idle: stop here as time isnt there, elapsed: 900
09-21 14:56:00.500 17847 19277 I DeltaChat: -------------------- FetchWorker.doWork() done --------------------
09-21 14:56:00.500 17847 17874 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=xxxxxxx-xxxxxxx-4745-bce5-827903f851fa, tags={ org.thoughtcrime.securesms.connect.FetchWorker } ]
09-21 15:11:00.670 17847 20322 I DeltaChat: -------------------- FetchWorker.doWork() started --------------------
09-21 15:11:00.670 17847 20322 I DeltaChat: ThreadId(27) src/job.rs           : 672: interrupt_inbox_idle: called -> from FetchWorker.doWork()
09-21 15:11:00.670 17847 20322 I DeltaChat: ThreadId(27) src/job.rs           : 691: interrupt_inbox_idle: stop here as time isnt there, elapsed: 1800
09-21 15:11:00.670 17847 20322 I DeltaChat: -------------------- FetchWorker.doWork() done --------------------
09-21 15:11:00.670 17847 17873 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=xxxxxxxxxx-xxxx-xxxx-bce5-827903f851fa, tags={ org.thoughtcrime.securesms.connect.FetchWorker } ]
09-21 15:12:26.580 17847 17879 I DeltaChat: ThreadId(5)  src/imap/idle.rs     : 106: Idle wait - has NewData
09-21 15:12:27.110 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 193: INBOX-IDLE ended...
09-21 15:12:27.110 17847 17879 I DeltaChat: ThreadId(5)  src/job.rs           :1065: perform_inbox_jobs: go ...
09-21 15:12:27.120 17847 17879 I DeltaChat: ThreadId(5)  src/job.rs           :1100: Imap job_perform: no job found => no action
09-21 15:12:27.120 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    :  76: JobThread::fetch INBOX-thread
09-21 15:12:27.120 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 104: JobThread::connect_and_fetch INBOX-thread
09-21 15:12:27.120 17847 17879 I DeltaChat: ThreadId(5)  src/context.rs       : 171: === network_online === is => true
09-21 15:12:27.120 17847 17879 I DeltaChat: ThreadId(5)  src/job_thread.rs    : 113: INBOX-fetch started...
09-21 15:12:27.120 17847 17879 I DeltaChat: ThreadId(5)  src/imap/mod.rs      : 594: Imap::fetch_new_messages
09-21 15:12:28.351 17847 17879 I DeltaChat: ThreadId(5)  src/context.rs       : 171: === network_online === is => true
09-21 15:12:29.562 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs:  42: Receiving message INBOX/43293...
09-21 15:12:29.702 17847 17879 I DeltaChat: ThreadId(5)  src/e2ee.rs          : 277: Detected Autocrypt-mime message
09-21 15:12:29.932 17847 17879 W DeltaChat: ThreadId(5)  src/mimeparser.rs    : 419: Ignoring nested protected headers
09-21 15:12:29.932 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs:1110: not creating ad-hoc group: too few contacts
09-21 15:12:30.002 17847 17879 I DeltaChat: ThreadId(5)  src/dc_receive_imf.rs: 649: Message has 1 parts and is assigned to chat #Chat#44.
09-21 15:12:30.002 17847 17879 I DeltaChat: ThreadId(5)  src/job.rs           :1384: Imap job_add: action MoveMsg, foreign_id 33252, delay_seconds 0
09-21 15:12:30.052 17847 17879 I DeltaChat: ThreadId(5)  src/job.rs           : 706: interrupt_inbox_idle_2 (without time check): called
09-21 15:12:30.052 17847 17879 W DeltaChat: ThreadId(5)  src/job.rs           : 728: interrupt_inbox_idle_2: could not interrupt idle: "WouldBlock", flag 
```

Reply:
Problem is, in some IDEs you can click on `src/job_thread.rs:111` in the log and directly get to this file and line. This won't work anymore now.

Reply:
agree, you could treat the file and lineno as one block and then pad it out however.  i think you'd still get your readability benefits.

(no pr is too small btw ;)

Reply:
Ok, makes sense. I wasn't aware of that.
But anyway will take this change out here and do an extra pr for that.